### ISSUE TITLE
False-positive `Variable is never used` is case of `if_let_guard`

### ISSUE THREAD

**Issue by mchernyavsky** [intellij-rust/intellij-rust#9500]
## Environment

* **IntelliJ Rust plugin version:** 0.4.180.4910-222-nightly
* **Rust toolchain version:** 1.66.0-nightly (2019147c5 2022-09-19) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.2.3 (CL-222.4167.35)
* **Operating system:** macOS 12.6
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description

<img width="678" alt="Screen Shot 2022-10-10 at 13 15 12" src="https://user-images.githubusercontent.com/6079006/194854083-fef1ab23-72fd-4372-b082-a9763c602cea.png">

Note that the plugin highlights one usage for the variable.

## Steps to reproduce
```rust
#![feature(if_let_guard)]

fn main() {
    match 42 {
        x if let x = x => { println!("{}", x) },
        _ => (),
    };
}

```

**Comment by Undin** [comment]
It seems #9354 should solve this issue
